plugin/gcp_config.go

```diff
@@ -67,25 +75,40 @@ func (c *gcpConfig) Update(d *framework.FieldData) error {
 
 	rawIamAlias, exists := d.GetOk("iam_alias")
 	if exists {
-		iamAlias := rawIamAlias.(string)
-		if iamAlias != c.IAMAliasType {
-			c.IAMAliasType = iamAlias
-		}
+		c.IAMAliasType = rawIamAlias.(string)
 	}
+
 	if err := c.IAMAuthMetadata.ParseAuthMetadata(d); err != nil {
-		return errwrap.Wrapf("failed to parse iam metadata: {{err}}", err)
+		return fmt.Errorf("failed to parse iam metadata: %w", err)
 	}
 
 	rawGceAlias, exists := d.GetOk("gce_alias")
 	if exists {
-		gceAlias := rawGceAlias.(string)
-		if gceAlias != c.GCEAliasType {
-			c.GCEAliasType = gceAlias
-		}
+		c.GCEAliasType = rawGceAlias.(string)
 	}
+
 	if err := c.GCEAuthMetadata.ParseAuthMetadata(d); err != nil {
-		return errwrap.Wrapf("failed to parse gce metadata: {{err}}", err)
+		return fmt.Errorf("failed to parse gce metadata: %w", err)
 	}
+
+	rawEndpoint, exists := d.GetOk("custom_endpoint")
+	if exists {
+		for k, v := range rawEndpoint.(map[string]string) {
```

Comment:
Wondering if there is any validation we should do on `v`. Looking closer at requirements for the `option.WithEndpoint()` that's provided as an option to clients.

Reply:
_If_ we did validation maybe just verify it is a valid URL? But I might lean more towards letting the user set what they want and allow the API call itself to report any errors.

Reply:
I tested out some bogus `option.WithEndpoint()` values, and Google's client doesn't validate that it's value has some specific format. So I'm +1 on letting the user set what they want here. The error will be apparent from the result of the API call. Thanks!